modules/iam-role-for-service-accounts-eks/main.tf

```diff
@@ -7,7 +7,7 @@ locals {
   partition           = data.aws_partition.current.partition
   dns_suffix          = data.aws_partition.current.dns_suffix
   region              = data.aws_region.current.name
-  role_name_condition = try(coalesce(var.role_name, "${var.role_name_prefix}*"), null)
+  role_name_condition = var.role_name != null ? var.role_name : "${var.role_name_prefix}*"
```

Comment:
this won't solve the issue - to solve the issue we just need to change the default value of `role_name_prefix` to `""`

Reply:
Setting the `role_name_prefix` to `""` does not solve the issue. It will create another issue because you will have `role_name` set to a value and `role_name_prefix` set to `""`, and providing both of them to `aws_iam_role` generates a conflict error:
`"name_prefix": conflicts with name`

Reply:
I think `aws_iam_role` can be created without either `name` or `name_prefix` which means both `role_name` and `role_name_prefix` can be kept  with default values `null`. In this case, "else" part will be executed and `"${var.role_name_prefix}*"` produces same error `Cannot include a null value in a string template.`
What do you think about changing exactly the problematic place? I mean, something like
```
role_name_condition = var.role_name != null ? var.role_name : "${var.role_name_prefix == null ? "" : var.role_name_prefix}*"
```
This way if role is created without name or prefix, `role_name_condition` is `*` but never null

Reply:
But we want to allow only this role to assume itself. 
```
condition {
        test     = "ArnLike"
        variable = "aws:PrincipalArn"
        values   = ["arn:${local.partition}:iam::${local.account_id}:role${var.role_path}${local.role_name_condition}"]
}
```
With `role_name_condition` set to `*,` the policy is too permissive.
The suggested solution solves the issue. Why is it problematic?

Reply:
Hm, you're right about too permissive. By problematic I meant the place that causes the error

Reply:
Yeah, I see. But what is wrong with the proposed solution?

Reply:
If module is initiated without specifying `role_name` and `role_name_prefix` they are `null` by default. So `var.role_name != null` is false and "else" statement, which is `"${var.role_name_prefix}*"` being executed. This produces error "Cannot include a null value in a string template."  even if `allow_self_assume_role` is false.

Reply:
In general, this solution is the same is it was in 5.19.0, so should be ok. 
Was I set as review on purpose or it's automatic?

Reply:
Yes, this solution is the same as it was in previous versions. And it is the same as how it is handled in other modules, such as: [iam-assumable-role-with-oidc/main.tf](https://github.com/terraform-aws-modules/terraform-aws-iam/blob/master/modules/iam-assumable-role-with-oidc/main.tf#L10)